interfaces/repo.go

```diff
@@ -656,3 +656,35 @@ func (r *Repository) RemoveSnap(snapInfo *snap.Info) error {
 
 	return nil
 }
+
+// DisconnectSnap disconnects all the connections to and from a given snap.
+//
+// The return value is a list of snap.Info's that were affected.
```

Comment:
Perhaps we disagree on how this is intended to be used:

The ``setup-security`` task will:
 - call repo.DisconnectSnap()
 - call repo.RemoveSnap()
 - call repo.AddSnap()
 - rebuild connections based on what is in the state (and update state if neccessary)
 - call backend.Setup() for each backend x affected snap

We cannot reconfigure just one snap. Even if we break this down into separate tasks, we have to know the affected set to properly react to snap changes.